packages/suite/src/actions/wallet/coinjoinAccountActions.ts

```diff
@@ -241,19 +242,83 @@ const coinjoinAccountCheckReorg =
             const txs = getAccountTransactions(
                 account.key,
                 state.wallet.transactions.transactions,
-            ).filter(({ blockHeight }) => !blockHeight || blockHeight >= checkpoint.blockHeight);
+            ).filter(
+                ({ blockHeight }) =>
+                    !blockHeight || blockHeight >= checkpoint.blockHeight || blockHeight < 0,
+            );
             dispatch(transactionsActions.removeTransaction({ account, txs }));
         }
     };
 
 const coinjoinAccountAddTransactions =
-    (account: Account, transactions: ScanAccountProgress['transactions']) =>
-    (dispatch: Dispatch) => {
-        if (transactions.length) {
-            dispatch(transactionsActions.addTransaction({ account, transactions }));
+    (props: Parameters<typeof transactionsActions.addTransaction>[0]) => (dispatch: Dispatch) => {
+        if (props.transactions.length > 0) {
+            dispatch(transactionsActions.addTransaction(props));
         }
     };
 
+/**
+Action called from coinjoinMiddleware as reaction to prepending tx creation.
+Prepending tx could be created either as result of successful CoinjoinRound (not broadcasted by suite)
+or as result of sendFormActions > addFakePendingTxThunk (broadcasted by suite)
+in both cases Account should:
+- exclude spent utxo
+- mark addresses as used
+- recalculate anonymity
+- recalculate balance
+prepending txs have deadline (blockheigth) when they should be removed from UI
```

Comment:
nitpick - typo in comment 
```suggestion
prepending txs have deadline (blockheight) when they should be removed from UI
```